frame/contracts/src/wasm/runtime.rs

```diff
@@ -96,6 +96,7 @@ impl KeyType {
 /// will not be changed or removed. This means that any contract **must not** exhaustively
 /// match return codes. Instead, contracts should prepare for unknown variants and deal with
 /// those errors gracefully in order to be forward compatible.
+#[derive(RuntimeDebug)]
```

Comment:
If you use `RuntimeDebug` the implementation is removed for `no_std` (replaced by the string `<stripped>`) builds to save space. But we want it to work on `no_std`.

You can try that by forcing the wasm runtime:
```
cargo run --release -- --dev -l "fatal,runtime::contracts::strace=trace" --execution=wasm
```

```suggestion
#[derive(Debug)]
```

Reply:
yes right, I was  not sure whether or not we wanted to use it with nostd, but got your point mentioned above that nostd is the way to go